providers/connection_mixin.py

```
@@ -0,0 +1,263 @@
+        Returns:
+            httpx.Client: An HTTPX client instance with connection pooling enabled.
+        """
+        # Create cache key based on configuration
+        cache_key = self._create_client_cache_key(base_url, timeout, headers)
+
+        with self._cache_lock:
+            # Return existing client if available
+            if cache_key in self._client_cache:
+                client = self._client_cache[cache_key]
+                if not client.is_closed:
+                    logger.debug(f"Reusing cached HTTP client for {base_url or 'default'}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)